File: java/Problem003.java

public class Problem003 {

	public static void main(String args[]) {
		System.out.println(new Problem003().run());
	}

	public String run() {
		long num = 600851475143L;
		int i = 2;

		while (i * i < num) {
			while (num % i == 0) {
				num /= i;
			}
			i++;
		}

		return "The largest prime factor of 600851475143 is " + num;

	}
}

File: python/Problem003.py
#6857.0 in 0.0020148754119873047
import time

start = time.time()

num = 600851475143  
i = 2

while i * i < num:
    while num % i == 0:
        num /= i
    i += 1

end = time.time()
print(num, "in", end-start)
